packages/apps/tools/src/components/Global/ModuleExplorer/index.tsx

```diff
@@ -1,89 +1,88 @@
-import React, { useCallback, useState } from 'react';
+import type { IncompleteModuleModel } from '@/hooks/use-module-query';
+import React, { useState } from 'react';
+import type { TreeItem } from '../CustomTree/CustomTree';
 import type { ISidePanelProps } from './SidePanel';
 import SidePanel from './SidePanel';
 import type { IEditorProps } from './editor';
 import Editor from './editor';
 import { containerStyle } from './styles.css';
-import type { IChainModule } from './types';
-import { getModulesMap } from './utils';
+import type { IChainModule, Outline } from './types';
+import { isModuleLike } from './types';
+import {
+  chainModuleToOutlineTreeItems,
+  moduleModelToChainModule,
+} from './utils';
 
 export interface IModuleExplorerProps {
-  modules: IChainModule[];
   openedModules: IEditorProps['openedModules'];
-  onModuleClick: ISidePanelProps['onResultClick'];
-  onInterfaceClick: ISidePanelProps['onInterfaceClick'];
-  onInterfacesExpand: ISidePanelProps['onInterfacesExpand'];
-  onModuleExpand: ISidePanelProps['onModuleExpand'];
+  onModuleClick: ISidePanelProps<IncompleteModuleModel>['onModuleClick'];
   onActiveModuleChange: IEditorProps['onActiveModuleChange'];
-  onTabClose: IEditorProps['onTabClose'];
+  // onTabClose: IEditorProps['onTabClose'];
```

Comment:
Do you still need this one?

Reply:
It will be enabled again in an upcoming PR